From f5e0365c8e76e2ef797027134f1819e613764d7e Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Wed, 19 Sep 2012 12:00:01 +0000
Subject: vf_lavc: fix compilation with latest libav

Remove a pointless and broken check for avctx->codec->encode.

1) The check does not test for anything useful.
2) AVCodecContext.encode is a private field and is not supposed to be
   accessed from outside of lavc.
2a) AVCodecContext.encode does not exist anymore in latest libavcodec,
    so this block fails to build.
---
diff --git a/libmpcodecs/vf_lavc.c b/libmpcodecs/vf_lavc.c
index b2c1dd7..65e93a1 100644
--- a/libmpcodecs/vf_lavc.c
+++ b/libmpcodecs/vf_lavc.c
@@ -76,11 +76,6 @@ static int config(struct vf_instance *vf,
 	return 0;
     }
 
-    if (lavc_venc_context.codec->encode == NULL) {
-	mp_msg(MSGT_VFILTER,MSGL_ERR,"avcodec init failed (ctx->codec->encode == NULL)!\n");
-	return 0;
-    }
-
     return vf_next_config(vf,width,height,d_width,d_height,flags,IMGFMT_MPEGPES);
 }
 
--
cgit v0.9
